### PR TITLE
5671 api token

### PR DESCRIPTION
**Pull request by lubitchv**
When a user use External tool in Configure button, the new API Token is generated. The message popups to confirm or cance. When token is generated an e-mail is sent to the user with a warning that it should be treated like a password.

The email, subject and the message  are added in Bundle.properties and will need to be translated to other languages. 

## Related Issues #5757 

- connects to #5671

## Pull Request Checklist

- [ ] Unit [tests][] completed
- [ ] Integration [tests][]: None
- [ ] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [ ] [Documentation][docs] completed
- [ ] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: http://guides.dataverse.org/en/latest/developers/sql-upgrade-scripts.html
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/7.3.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

File: src/main/java/edu/harvard/iq/dataverse/ConfigureFragmentBean.java [hunk 1]
```diff
@@ -35,15 +41,21 @@
...
     @EJB
     AuthenticationServiceBean authService;
+    @EJB
+    UserNotificationServiceBean userNotificationService;
     
-    public String configureExternalAlert() { 
+    public String configureExternalAlert() {
+        generateApiToken();
+        String httpString = "window.open('" + toolHandler.getToolUrlWithQueryParams()+  "','_blank'" +")";
+        RequestContext.getCurrentInstance().execute(httpString);
```
Comment (qqmyers) [inline src/main/java/edu/harvard/iq/dataverse/ConfigureFragmentBean.java]: I think this is deprecated. 
PrimeFaces.current().executeScript(httpString);
should do the same thing...

Comment (lubitchv) [reply src/main/java/edu/harvard/iq/dataverse/ConfigureFragmentBean.java]: You are right. I replaced RequestContext with PrimeFaces as you recomended.

File: src/main/java/edu/harvard/iq/dataverse/ConfigureFragmentBean.java [hunk 2]
```diff
@@ -55,9 +51,9 @@
     
     public String configureExternalAlert() {
         generateApiToken();
+        PrimeFaces.current().executeScript("location.reload(true)");
```
Comment (scolapasta) [inline src/main/java/edu/harvard/iq/dataverse/ConfigureFragmentBean.java]: @lubitchv could you briefly explain what this change does?

Comment (lubitchv) [reply src/main/java/edu/harvard/iq/dataverse/ConfigureFragmentBean.java]: It reloads the dataverse page. Without reloading the page one would not see the notification. There is also another reason for that. If user has several files in DataSet and he uses external tool for the first file and later returns for the page to use external tool for the second file, the fileId will remain of the first file end external tool will show the user content of the first file and not of a second. It confuses users tremendously. We had an experience that many of users do not pay attention to the message that asks them to reload the browser. So I thought it is better to do for them automagically.

Comment (scolapasta) [reply src/main/java/edu/harvard/iq/dataverse/ConfigureFragmentBean.java]: @lubitchv OK, we'll have @kcondon test that and see how it works. Can you also then remove the message from the bundle (I'm assuming, of course, it was only used in this one place)?

**Pull request merged**